pkg/docker/util.go

```diff
@@ -155,3 +159,85 @@ func parseRepositoryTag(repos string) (string, string, string) {
 	}
 	return repos, "", ""
 }
+
+// PullImage pulls the Docker image specifies by name taking the pull policy
+// into the account.
+// TODO: The 'force' option will be removed
+func PullImage(name string, d Docker, policy api.PullPolicy, force bool) (*PullResult, error) {
+	// TODO: Remove this after we deprecate --force-pull
+	if force {
+		policy = api.PullAlways
+	}
+
+	// When pull policy not specified, default to DefaultPullPolicy
+	if len(policy) == 0 {
+		policy = api.DefaultBuilderPullPolicy
```

Comment:
you're defaulting to the Builder policy, but this method might be invoked for Builder images or Previous images...  that's a problem.  You should just require a policy be passed in and error out otherwise.